### PR TITLE
Reporting errors of the preprocess that is the second property of object

### PR DESCRIPTION
**Pull request by yukukotani**
Fix https://github.com/colinhacks/zod/issues/2904

This behavior was introduced in #2426. `ctx.common.issues` includes the errors of previous properties, so that they affects the status of preprocess.

### CODE REVIEW

**Comment by yukukotani** [comment]
I'm not confident in my understanding of the whole codebase and the change in #2426.

@Pingviinituutti Could you take a look at this PR?

File: deno/lib/__tests__/transformer.test.ts [hunk 1]
```diff
@@ -297,6 +297,23 @@ test("short circuit on dirty", () => {
...
+    nonEmptyStr: z.string().min(1),
+    positiveNum: z.preprocess((v) => Number(v), z.number().positive()),
+  });
+  const result = schema.safeParse({
+    nonEmptyStr: "",
+    positiveNum: "",
+  });
+  expect(result.success).toEqual(false);
+  if (!result.success) {
+    expect(result.error.issues.length).toEqual(2);
```
Comment (yukukotani) [inline deno/lib/__tests__/transformer.test.ts:311]: This fails in current master branch since `result.error.issues.length === 1`

**Comment by Pingviinituutti** [comment]
Looks good to me, seems the simple fix is enough to :)

Are the current test cases enough to ensure it works as intended?

**Comment by yukukotani** [comment]
@Pingviinituutti Thanks!

> Are the current test cases enough to ensure it works as intended?

The current test cases are fail on master branch, so I think it's minimum but enough.

**Comment by yukukotani** [comment]
@Austin-Love This is ready and waiting for getting review

**Comment by colinhacks** [comment]
Made some fairly major changes to `ctx.addIssue` in `preprocess` actually agrees with the runtime behavior in `.superRefine` and `.transform`. Namely, all issues are considered *non-fatal* — validation will continue even if `.addIssue` was called during the `preprocess`. This behavior can be modified by indicating `fatal: true` in `ctx.addIssue`:

```ts
const schema = z.preprocess((data, ctx) => {
      ctx.addIssue({
        code: "custom",
        message: `bad thing happened`,
        fatal: true
      });
      return data;
    }, z.string())
```

**Pull request merged**